engine/runtime/src/main/java/org/enso/interpreter/runtime/data/text/Text.java

```diff
@@ -193,19 +193,58 @@ String toDisplayString(
       boolean allowSideEffects,
       @Cached("build()") @Cached.Shared("strings") ToJavaStringNode toJavaStringNode) {
     String str = toJavaStringNode.execute(this);
-    // TODO This should be more extensible
-    String replaced =
-        str.replace("\\", "\\\\")
-            .replace("'", "\\'")
-            .replace("\n", "\\n")
-            .replace("\t", "\\t")
-            .replace("\u0007", "\\a")
-            .replace("\u0008", "\\b")
-            .replace("\u000c", "\\f")
-            .replace("\r", "\\r")
-            .replace("\u000B", "\\v")
-            .replace("\u001B", "\\e");
-    return "'" + replaced + "'";
+    int len = str.length();
+    int outputLength = len + 2; // Precise if there are no special characters.
+
+    // TODO This should be more extensible; while it's still a small fixed set,
```

Comment:
Do we need this comment? It says `TODO`, but I am not sure what shall we do?

Reply:
The comment was "This should be more extensible", which I assume means that we'd have a table of special characters and their translations, but I'm happy to take it out.